crates/l2/proposer/l1_watcher.rs

```diff
@@ -69,6 +44,46 @@ impl L1Watcher {
             max_block_step: watcher_config.max_block_step,
             last_block_fetched: U256::zero(),
             l2_proposer_pk: watcher_config.l2_proposer_private_key,
+            check_interval: Duration::from_millis(watcher_config.check_interval_ms),
+        }
+    }
+
+    pub async fn run(&mut self, store: &Store) {
+        loop {
+            if let Err(err) = self.main_logic(store.clone()).await {
+                error!("L1 Watcher Error: {}", err);
+            }
+
+            sleep(Duration::from_millis(200)).await;
```

Comment:
sure, working on it.

Reply:
Fixed, used the `check_interval` for the `run` loop. The sleep in the main_logic may be needed. Since it would be asking the contract faster than needed.